src/ServiceControl.Persistence.RavenDb/Throughput/ThroughputDataStore.cs

```diff
@@ -0,0 +1,287 @@
+using Raven.Client.Documents.Queries;
+using Raven.Client.Documents.Session;
+
+class ThroughputDataStore(
+    IRavenDocumentStoreProvider storeProvider,
+    ThroughputDatabaseConfiguration databaseConfiguration) : ILicensingDataStore
+{
+    internal const string ThroughputTimeSeriesName = "INC: throughput data";
+    const string AuditServiceMetadataDocumentId = "AuditServiceMetadata";
+    const string BrokerMetadataDocumentId = "BrokerMetadata";
+    const string ReportMasksDocumentId = "ReportMasks";
+
+    static readonly AuditServiceMetadata DefaultAuditServiceMetadata = new([], []);
+    static readonly BrokerMetadata DefaultBrokerMetadata = new(null, []);
+    static readonly ReportConfigurationDocument DefaultReportConfiguration = new();
+
+    public async Task<IEnumerable<Endpoint>> GetAllEndpoints(bool includePlatformEndpoints, CancellationToken cancellationToken)
+    {
+        var store = await storeProvider.GetDocumentStore(cancellationToken);
+        using IAsyncDocumentSession session = store.OpenAsyncSession(databaseConfiguration.Name);
+
+        var baseQuery = session.Query<EndpointDocument>();
+
+        var query = includePlatformEndpoints
+            ? baseQuery
+            : baseQuery.Where(document => !document.EndpointIndicators.Contains(EndpointIndicator.PlatformEndpoint.ToString()));
+
+        var documents = await query.ToListAsync(cancellationToken);
+
+        return documents.Select(document => document.ToEndpoint());
+    }
+
+    public async Task<Endpoint?> GetEndpoint(EndpointIdentifier id, CancellationToken cancellationToken)
+    {
+        var store = await storeProvider.GetDocumentStore(cancellationToken);
+        using IAsyncDocumentSession session = store.OpenAsyncSession(databaseConfiguration.Name);
+
+        var documentId = id.GenerateDocumentId();
+
+        var document = await session.LoadAsync<EndpointDocument>(
+            documentId,
+            builder => builder.IncludeTimeSeries(ThroughputTimeSeriesName),
+            cancellationToken);
+
+        var endpoint = document?.ToEndpoint();
+        if (endpoint != null)
+        {
+            var timeSeries = await session
+                .IncrementalTimeSeriesFor(documentId, ThroughputTimeSeriesName)
+                .GetAsync(token: cancellationToken);
+
+            endpoint.LastCollectedDate = DateOnly.FromDateTime(timeSeries.LastOrDefault()?.Timestamp ?? DateTime.MinValue);
+        }
+
+        return endpoint;
+    }
+
+    public async Task<IEnumerable<(EndpointIdentifier, Endpoint?)>> GetEndpoints(IEnumerable<EndpointIdentifier> endpointIds, CancellationToken cancellationToken)
+    {
+        var endpoints = new List<(EndpointIdentifier, Endpoint?)>();
+
+        var store = await storeProvider.GetDocumentStore(cancellationToken);
+        using IAsyncDocumentSession session = store.OpenAsyncSession(databaseConfiguration.Name);
+
+        var documentIdLookup = endpointIds.ToDictionary(
```

Comment:
Possible multiple enumeration of `endpointIds` ?

Reply:
Fixed